core/src/main/java/kafka/server/share/SharePartition.java

```diff
@@ -380,6 +381,7 @@ public CompletableFuture<Void> maybeInitialize() {
         // All the pending requests should wait to get completed before the share partition is initialized.
         // Attain lock to avoid any concurrent requests to be processed.
         lock.writeLock().lock();
+        AtomicReference<Optional<Throwable>> futureException = new AtomicReference<>(Optional.empty());
```

Comment:
I'm not convinced that you need an `Optional` in here. This gives you three options: null, Optional.empty, and a non-empty Optional. If you just need two, wouldn't `Optional<Throwable>` be slightly less cumbersome?

Reply:
I need to use an `AtomicReference` here since only final or effective final variables can be passed into `whenComplete` block of `persister.readState`. The only reason for using AtomicReference<Optional<Throwable>> and not AtomicReference<Throwable> is because the exception could be null as well. Hence, thought using Optional is a better coding standard here.
Regarding the 3 options - null, Optional.empty, and a non-empty Optional, I initialize futureException to Optional.empty() and never use null. Hence, only dealing with 2 option values here.

I can go ahead and make the change if my reasoning is not satisfactory.

Reply:
I'm perfectly happy with your reasoning. I will just push on this kind of thing to make sure you meant it. Leave it as it is.